h2o-r/h2o-package/R/gam.R

```diff
@@ -0,0 +1,694 @@
+#' @param lambda_min_ratio Minimum lambda used in lambda search, specified as a ratio of lambda_max (the smallest lambda that drives all
+#'        coefficients to zero). Default indicates: if the number of observations is greater than the number of
+#'        variables, then lambda_min_ratio is set to 0.0001; if the number of observations is less than the number of
+#'        variables, then lambda_min_ratio is set to 0.01. Defaults to -1.
+#' @param beta_constraints Beta constraints
+#' @param max_active_predictors Maximum number of active predictors during computation. Use as a stopping criterion to prevent expensive model
+#'        building with many predictors. Default indicates: If the IRLSM solver is used, the value of
+#'        max_active_predictors is set to 5000 otherwise it is set to 100000000. Defaults to -1.
+#' @param interactions A list of predictor column indices to interact. All pairwise combinations will be computed for the list.
+#' @param interaction_pairs A list of pairwise (first order) column interactions.
+#' @param obj_reg Likelihood divider in objective value computation, default is 1/nobs Defaults to -1.
+#' @param export_checkpoints_dir Automatically export generated models to this directory.
+#' @param balance_classes \code{Logical}. Balance training data class counts via over/under-sampling (for imbalanced data). Defaults to
+#'        FALSE.
+#' @param class_sampling_factors Desired over/under-sampling ratios per class (in lexicographic order). If not specified, sampling factors will
+#'        be automatically computed to obtain class balance during training. Requires balance_classes.
+#' @param max_after_balance_size Maximum relative size of the training data after balancing class counts (can be less than 1.0). Requires
+#'        balance_classes. Defaults to 5.0.
+#' @param max_hit_ratio_k Maximum number (top K) of predictions to use for hit ratio computation (for multi-class only, 0 to disable)
+#'        Defaults to 0.
+#' @param max_runtime_secs Maximum allowed runtime in seconds for model training. Use 0 to disable. Defaults to 0.
+#' @param custom_metric_func Reference to custom evaluation function, format: `language:keyName=funcName`
+#' @param k Number of knots for gam predictors
+#' @param knots_keys String arrays storing frame keys of knots.  One for each gam column specified in gam_x
+#' @param bs Basis function type for each gam predictors, 0 for cr
+#' @param scale Smoothing parameter for gam predictors
+#' @param save_gam_cols \code{Logical}. Save keys of model matrix Defaults to FALSE.
+#' @param segment_columns A list of columns to segment-by. H2O will group the training (and validation) dataset by the segment-by columns
+#'        and train a separate model for each segment (group of rows).
+#' @param segment_models_id Identifier for the returned collection of Segment Models. If not specified it will be automatically generated.
+#' @export
+h2o.bulk_gam <- function(x,
```

Comment:
@wendycwong Is there a precedent in any other functions in H2O to train K models for K segments of the training set?  If so, can you point me there?  I am assuming the reason that you created a separate function for this `h2o.bulk_gam()` is because the output is different?  It looks like a list of K models.  I just want to make sure we are being consistent with existing structures and don't define a secondary specialized function if we don't have to.